tools/mirror/src/lib.rs

```diff
@@ -1747,13 +1961,55 @@ impl<T: ChainAccess> TxMirror<T> {
 
         tracing::debug!(target: "mirror", "source chain initialized with first heights: {:?}", &next_heights);
 
-        let mut tracker = crate::chain_tracker::TxTracker::new(
+        let tracker = Arc::new(Mutex::new(crate::chain_tracker::TxTracker::new(
             self.target_min_block_production_delay,
             self.config.tx_batch_interval,
             next_heights.iter(),
             stop_height,
-        );
-        let (target_height, target_head) = self.index_target_chain(&mut tracker).await?;
+        )));
+        let target_height = Arc::new(RwLock::new(0));
+        let target_head = Arc::new(RwLock::new(CryptoHash::default()));
+        let (clients_tx, clients_rx) = tokio::sync::oneshot::channel();
+        let (target_indexer_done_tx, target_indexer_done_rx) =
+            tokio::sync::oneshot::channel::<anyhow::Result<()>>();
+        let (unstake_tx, unstake_rx) = mpsc::channel(10);
+
+        let db = self.db.clone();
+        let target_height2 = target_height.clone();
+        let target_head2 = target_head.clone();
+        let tracker2 = tracker.clone();
+        let arbiter = actix::Arbiter::new();
```

Comment:
Can you use more descriptive names here?

Reply:
Are you referring to the naming pattern `let X2 = X.clone()`, or the names `X` themselves? Do you have suggestions for what you had in mind? Like `target_head` referring to the head hash of the target chain already seems kind of descriptive to me, no? If you have suggestions, I'm happy to hear them though

Reply:
renamed the arbiters, in case thats what you meant